### ISSUE TITLE
Add a sanitize package

### ISSUE THREAD

**Issue by ntwb** [WordPress/gutenberg#7824]
@adamsilverstein commented on [Tue Jul 18 2017](https://github.com/WordPress/packages/pull/12)

These could use some unit tests.

---

@nylen commented on [Tue Jul 25 2017](https://github.com/WordPress/packages/pull/12#issuecomment-317544687)

When are these functions expected to be used?  In particular the combination of "strip tags and encode HTML entities" seems a bit strange to me.

---

@ntwb commented on [Wed Aug 16 2017](https://github.com/WordPress/packages/pull/12#issuecomment-322700654)

I've added a handful of basic tests for `stripTags` via https://github.com/WordPress/packages/pull/12/commits/7e360f0fa7f5982d9833afbf28238eecb6727639

---

@codecov[bot] commented on [Fri Aug 18 2017](https://github.com/WordPress/packages/pull/12#issuecomment-322700958)

# [Codecov](https://codecov.io/gh/WordPress/packages/pull/12?src=pr&el=h1) Report
> Merging [#12](https://codecov.io/gh/WordPress/packages/pull/12?src=pr&el=desc) into [master](https://codecov.io/gh/WordPress/packages/commit/d7c1b5e2f5bc9e12953e167a593da77b320c9b42?src=pr&el=desc) will **decrease** coverage by `12.06%`.
> The diff coverage is `30%`.

[![Impacted file tree graph](https://codecov.io/gh/WordPress/packages/pull/12/graphs/tree.svg?width=650&height=150&src=pr&token=9Zw9xmxPWW)](https://codecov.io/gh/WordPress/packages/pull/12?src=pr&el=tree)

```diff
@@             Coverage Diff             @@
##           master      #12       +/-   ##
===========================================
- Coverage     100%   87.93%   -12.07%     
===========================================
  Files           6        9        +3     
  Lines          48       58       +10     
  Branches        7       10        +3     
===========================================
+ Hits           48       51        +3     
- Misses          0        5        +5     
- Partials        0        2        +2
```

| [Impacted Files](https://codecov.io/gh/WordPress/packages/pull/12?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [packages/sanitize/src/sanitizeText.js](https://codecov.io/gh/WordPress/packages/pull/12?src=pr&el=tree#diff-cGFja2FnZXMvc2FuaXRpemUvc3JjL3Nhbml0aXplVGV4dC5qcw==) | `0% <0%> (ø)` | |
| [packages/sanitize/src/index.js](https://codecov.io/gh/WordPress/packages/pull/12?src=pr&el=tree#diff-cGFja2FnZXMvc2FuaXRpemUvc3JjL2luZGV4Lmpz) | `0% <0%> (ø)` | |
| [packages/sanitize/src/stripTags.js](https://codecov.io/gh/WordPress/packages/pull/12?src=pr&el=tree#diff-cGFja2FnZXMvc2FuaXRpemUvc3JjL3N0cmlwVGFncy5qcw==) | `100% <100%> (ø)` | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/WordPress/packages/pull/12?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/WordPress/packages/pull/12?src=pr&el=footer). Last update [d7c1b5e...6c93f80](https://codecov.io/gh/WordPress/packages/pull/12?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

---

@adamsilverstein commented on [Fri Aug 18 2017](https://github.com/WordPress/packages/pull/12#issuecomment-323153811)

> When are these functions expected to be used? In particular the combination of "strip tags and encode HTML entities" seems a bit strange to me.
@nylen -
In WordPress core, the functions are used in wp-admin/js/press-this.js. `sanitizeText` is called in `getTitleText` to safely set the untrusted title text: `$( '#post_title' ).val( getTitleText() );`. `stripTags` is used in that function and also might be generally useful: it uses regexs to strip tags from content.

---

@notnownikki commented on [Mon Aug 21 2017](https://github.com/WordPress/packages/pull/12#issuecomment-323696676)

I just added a `decodeEntities` util to Gutenberg, and it was suggested it might be relevant here too.

Commit is at https://github.com/WordPress/gutenberg/commit/f08c5f46b690e7bd0b53aa8ac93169e3a3681ec9 , if it's useful I'm happy to work on contributing it here.

---

@adamsilverstein commented on [Tue Oct 10 2017](https://github.com/WordPress/packages/pull/12#issuecomment-335152862)

@notnownikki that does seem useful, not sure it belongs with `sanitize`? I had originally put these helpers in wp.utils, only later separating them out.

**Comment by aduth** [comment]
@adamsilverstein I'm trying to reinterpret the requirements from https://github.com/WordPress/packages/pull/12 , which mentions `press-this.js`.

Should we consider a package if / when the time comes we decide to port that into the mono-repository? Ties back into some of the conversation in [yesterday's JavaScript chat](https://make.wordpress.org/core/2019/03/19/javascript-chat-summary-march-19-2019/) about not proactively porting modules.

There's also a combination of `escape-html` (introduced by @ellatrix in #7890) and `html-entities` (introduced by @gziolo in #7977) which can cover some of the proposed requirements.

tl;dr Can we close this?

**Comment by adamsilverstein** [comment]
This was related to extracting existing core functionality on `wp.sanitize` into a package (see https://github.com/WordPress/wordpress-develop/blob/master/src/js/_enqueues/wp/sanitize.js). Not sure how much value these have or how they compare to the helpers you mentioned.

I only found one place where we actually use these functions in core: https://github.com/WordPress/wordpress-develop/blob/267c79a5182b3eebdb319874bb5e3ae168a9a6ac/src/js/_enqueues/wp/theme-plugin-editor.js#L119 (although there are probably some other places where we _should_ be using these. As you mentioned, these functions originally came from PressThis which has now been removed from core entirely. 

tl;dr; Yea, sure - we can go ahead and close this.

**Comment by aduth** [comment]
Thanks for checking it out, @adamsilverstein
